rust/lance-index/src/vector/v3/shuffler.rs

```diff
@@ -248,6 +248,10 @@ impl ShuffleReader for IvfShufflerReader {
         &self,
         partition_id: usize,
     ) -> Result<Option<Box<dyn RecordBatchStream + Unpin + 'static>>> {
+        if partition_id >= self.partition_sizes.len() {
+            return Ok(None);
+        }
```

Comment:
similar question here - is it a cause for concern that this function would be asked for an out of bounds partition? If so should this be an error?